website/src/Pages/ResourcesPage/AddResourcePage.jsx

```diff
@@ -0,0 +1,286 @@
+  const [longitude, setLongitude] = useState('');
+  const [address, setAddress] = useState('');
+  const [email, setEmail] = useState('');
+  const [website, setWebsite] = useState('');
+
+  const backToResources = () => {
+    navigate('/resources');
+  };
+
+  const addDayTime = () => {
+    const newDayTimes = [...dayTimes];
+    newDayTimes.push({
+      day: 0,
+      startTime: dayjs(),
+      endTime: dayjs().add(1, 'h'),
+    });
+    setDayTimes(newDayTimes);
+  };
+
+  const removeDayTime = (index) => {
+    const newDayTimes = [...dayTimes];
+    newDayTimes.splice(index, 1);
+    setDayTimes(newDayTimes);
+  };
+
+  const handlePhoneNumberChange = (event) => {
+    const inputValue = event.target.value;
+
+    // Remove all non-numeric characters from the input value
+    const numericValue = inputValue.replace(/\D/g, '');
+
+    // Format the phone number once it reaches 10 digits
+    if (numericValue.length === 10) {
+      const formattedValue = `(${numericValue.slice(0, 3)}) ${numericValue.slice(3, 6)}-${numericValue.slice(6, 10)}`;
+      setPhoneNumber(formattedValue);
+    } else if (numericValue.length < 10) {
+      setPhoneNumber(numericValue);
+    }
+  };
+
+  const validEmail = ((e) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(e);
+  });
+
+  function validWebsite(url) {
+    const urlPattern = /^(https?:\/\/)?([\da-z.-]+)\.([a-z.]{2,6})([/\w .-]*)*\/?$/i;
+    return urlPattern.test(url);
+  }
+
+  const validateResourceInputs = () => (title === '' || tag === ''
+  || latitude === '' || longitude === '' || address === '' || !validEmail(email) || !validWebsite(website));
+
+  const groupDayTimes = () => {
+    const groupedMap = new Map();
+
+    // Iterate through the dayTimeArray
+    dayTimes.forEach((dayTime) => {
+      const startTimeString = dayTime.startTime.format('h:mm a');
+      const endTimeString = dayTime.endTime.format('h:mm a');
+      const key = `${startTimeString}-${endTimeString}`;
+
+      if (groupedMap.has(key)) {
+        const existingGroupedDayTime = groupedMap.get(key);
+        const updatedGroupedDayTime = {
+          ...existingGroupedDayTime,
+          days: [...new Set([...existingGroupedDayTime.days, dayTime.day])].sort((a, b) => a - b),
+        };
+        groupedMap.set(key, updatedGroupedDayTime);
+      } else {
+        groupedMap.set(key, {
+          days: [dayTime.day],
+          startTime: startTimeString,
+          endTime: endTimeString,
+        });
+      }
+    });
+
+    // Create an array of grouped objects
+    const groupedResults = Array.from(groupedMap.values());
+
+    return groupedResults;
+  };
+
+  const createResource = async () => {
+    if (validateResourceInputs()) {
+      alert('Missing field or incorrect format.');
+      return;
+    }
+
+    const dateDetails = groupDayTimes();
+    const locationDetails = {
+      address,
+      coordinates: {
+        latitude: Number(latitude),
```

Comment:
not sure if this is sufficient validation, as when you give this a non-numeric string, latitude becomes NaN and then the server will send a 400 Bad Request reply after making the API call